source/configure/calls-deployment.rst

```diff
@@ -0,0 +1,187 @@
+Calls self-hosted deployment
+============================
+
+This document provides information on how to successfully make the Calls plugin work on self-hosted deployments. It also outlines some of the most common deployment strategies with example diagrams.
+
+Components
+----------
+
+- **Calls plugin**: This is the main entry point and a requirement to enable channel calls. 
+- **RTCD**: This is an optional service that can be deployed to offload all the functionality and data processing involved with the WebRTC connections. This is the preferred solution for a performant and scalable deployment. With RTCD, the Mattermost server will be minimally impacted when hosting a high number of calls.
+
+Terminology
+-----------
+
+- **WebRTC**: The set of underlying protocols/specifications on top of which calls are implemented. 
+- **RTC (Real Time Connection)**: The real-time connection. This is the channel used to send media tracks (audio/video/screen).
+- **WS (WebSocket)**: The WebSocket connection. This is the channel used to set up a connection (signaling process).
+- **NAT (Network Address Translation)**: A networking technique to map IP addresses. 
+- **STUN (Session Traversal Utilities for NAT)**: A protocol/service used by WebRTC clients to help traversing NATs. On the server side it's mainly used to figure out the public IP of the instance. 
+- **TURN (Traversal Using Relays around NAT)**: A protocol/service used to help WebRTC clients behind strict firewalls connect to a call through media relay. 
```

Comment:
We had some links in the original document that explained more in details what some of these protocols are about. Do we have a policy against linking to external material or is it the specific source we are worried about? I can provide Wikipedia articles if you prefer, they are just a bit lower level and harder to digest for less technical people.

Reply:
No that's my bad, apologies! I'll re-link them. Thanks for catching that. :)